### ISSUE TITLE
Buttons consume all button events regardless of button mask

### ISSUE THREAD

**Issue by Ryhon0** [godotengine/godot#89143]
### Tested versions

4.2.1 mono

### System information

Godot v4.2.1.stable.mono (b09f793f5) - Arch Linux #1 ZEN SMP PREEMPT_DYNAMIC Sat, 24 Feb 2024 04:03:29 +0000 - X11 - GLES3 (Compatibility) - NVIDIA GeForce GTX 1050 Ti (nvidia; 550.54.14) - AMD Ryzen 5 3600 6-Core Processor (12 Threads)

### Issue description

When the cursor is over a button and any mouse button is pressed, that event becomes consumed by the button and not passed to the parent, e.g. an item in a list has a delete button, which should only activate with the left button, right clicking anywhere on the item should open a popup menu with options but right clicking with the cursor on the delete button consumes the input. The mouse filter for the delete button may not be set to pass because it would activate both the delete button and the list item.

### Steps to reproduce

* Create a node, which handles right click inputs
* Add a button as a child of it with button mask set to mouse left
* Run the scene and try to right click the parent control with the cursor over the button

### Minimal reproduction project (MRP)

This project shows the example described in the project description.
[buttonmask.zip](https://github.com/godotengine/godot/files/14482239/buttonmask.zip)

**Comment by AThousandShips** [comment]
Buttons don't accept events, so unsure what's going on here

**Comment by Sauermann** [comment]
In your MRP, please try to change `mouse_filter` from `Stop` to `Pass`.
That should make your MRP work, as you expect.
`mouse_filter == Pass` implies, that the event is passed to the parent node, while `Stop` doesn't.

**Comment by Ryhon0** [comment]
Left clicking on the delete button with filter set to pass causes both of the buttons to register a press. I'd expect the delete button to consume the left button and pass any other

**Comment by AThousandShips** [comment]
It doesn't, as I said above it doesn't consume events

**Comment by Sauermann** [comment]
Can you please try this project: [buttonmask.zip](https://github.com/godotengine/godot/files/14486079/buttonmask.zip)

I have made some adjustments to your project and believe, that with these changes it should work.
The trick is not to connect `queue_free` directly to the signal, but to connect a function, that calls both `queue_free` and `accept_event`.

**Comment by Ryhon0** [comment]
Yeah, this works. Connecting the up and down signals to accept_event does the job but doing this for an entire project could be tedious, I'd expect this to be an engine feature.
